pilot/pkg/serviceregistry/kube/controller/ambient/policies.go

```diff
@@ -59,6 +62,23 @@ func WaypointPolicyStatusCollection(
 				reason := "unknown"
 				bound := false
 				switch target.GetKind() {
+				case gvk.GatewayClass_v1.Kind:
+					fetchedGatewayClasses := krt.Fetch(ctx, gatewayClasses, krt.FilterKey(target.GetName()))
+					if len(fetchedGatewayClasses) == 1 {
+						// verify GatewayClass is for waypoint
+						if fetchedGatewayClasses[0].Spec.ControllerName != constants.ManagedGatewayMeshController {
+							reason = model.WaypointPolicyReasonInvalid
+							message = fmt.Sprintf("non-waypoint GatewayClass `%s` specified, GatewayClass must use controller name `%s`", fetchedGatewayClasses[0].GetName(), constants.ManagedGatewayMeshController)
+						} else {
+							bound = true
+							reason = model.WaypointPolicyReasonAccepted
+							message = fmt.Sprintf("bound to %s", fetchedGatewayClasses[0].GetName())
+						}
+					} else if len(fetchedGatewayClasses) > 1 {
```

Comment:
This should be impossible, to have multiple GatewayClass with the same name, I would just switch to FetchOne

Reply:
if it is impossible then i will remove!